framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -12,18 +12,433 @@
+		const isChainLive = await interoperabilityStore.isLive(partnerChainIDBuffer, Date.now());
+		if (partnerChainAccount.status === CHAIN_ACTIVE && !isChainLive) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Sending partner chain ${txParams.sendingChainID} is not live.`),
+			};
+		}
+
+		// Liveness Requirement for the First CCU
+		if (partnerChainAccount.status === CHAIN_REGISTERED) {
+			// Certificate must not be empty for first CCU
+			if (!txParams.certificate) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error(
+						`Sending partner chain ${txParams.sendingChainID} is in registered status so certificate cannot be empty.`,
+					),
+				};
+			}
+		}
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+		const invalidCertificateErrors = validator.validate(certificateSchema, decodedCertificate);
+		if (invalidCertificateErrors.length > 0) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate does not follow valid certificate schema.'),
+			};
+		}
+		if (
+			txParams.activeValidatorsUpdate &&
+			txParams.activeValidatorsUpdate.length !== 0 &&
+			txParams.newCertificateThreshold > BigInt(0)
+		) {
+			// Non-empty certificate
+			if (!txParams.certificate) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Certificate cannot be empty.'),
+				};
+			}
+
+			// params.activeValidatorsUpdate has the correct format
+			for (let i = 0; i < txParams.activeValidatorsUpdate.length; i += 1) {
+				const currentValidator = txParams.activeValidatorsUpdate[i];
+				const nextValidator = txParams.activeValidatorsUpdate[i + 1];
+				if (currentValidator.blsKey.byteLength !== VALID_BLS_KEY_LENGTH) {
+					return {
+						status: VerifyStatus.FAIL,
+						error: new Error(`BlsKey length should be equal to ${VALID_BLS_KEY_LENGTH}.`),
+					};
+				}
+				if (currentValidator.blsKey.compare(nextValidator.blsKey) > -1) {
+					return {
+						status: VerifyStatus.FAIL,
+						error: new Error('Validators blsKeys must be unique and lexicographically ordered'),
+					};
+				}
+			}
+		}
+
+		const partnerChannelStore = context.getStore(transaction.moduleID, STORE_PREFIX_CHANNEL_DATA);
+		const partnerChannelData = await partnerChannelStore.getWithSchema<ChannelData>(
+			partnerChainIDBuffer,
+			channelSchema,
+		);
+		// InboxUpdate Validity
+		const { crossChainMessages, messageWitness, outboxRootWitness } = txParams.inboxUpdate;
+		const ccmHashes = crossChainMessages.map(ccm => hash(ccm));
+
+		let newInboxRoot;
+		let newInboxAppendPath;
+		let newInboxSize;
+		for (const ccm of ccmHashes) {
+			const { appendPath, size } = regularMerkleTree.calculateMerkleRoot({
+				value: ccm,
+				appendPath: partnerChannelData.inbox.appendPath,
+				size: partnerChannelData.inbox.size,
+			});
+			newInboxAppendPath = appendPath;
+			newInboxSize = size;
```

Comment:
`newInboxRoot` also should be assigned here?

Reply:
If I assign then it will throw an error in deepscan coz its value is not used https://deepscan.io/dashboard/#view=project&tid=6759&pid=8869&bid=554464&subview=pull-request&prid=1233154 , we assign it later after which it is used